web/modules/custom/grundsalg_db_client/src/Controller/ApiController.php

```diff
@@ -45,21 +85,123 @@ public function subdivisionUpdated($sid) {
       }
 
       // Create the content.
-      $contentCreationService = \Drupal::service('grundsalg.content_creation');
-      $operation = $contentCreationService->updateSubdivision((array) $content);
+      $operation = $this->contentCreationService->updateSubdivision((array)$content);
 
       return new JsonResponse([
-        'error' => FALSE,
+        'error' => false,
         'message' => $operation,
         'body' => $body,
       ]);
+    } catch (\Throwable $throwable) {
+      return $this->error($throwable);
     }
-    catch (\Exception $e) {
-      return new JsonResponse([
-        'error' => TRUE,
-        'message' => $e->getMessage(),
-        'body' => $e->getMessage(),
-      ]);
+  }
+
+  public function grunde($sid)
+  {
+    try {
+      $request = $this->requestStack->getCurrentRequest();
+
+      $path = sprintf('%s/grunde.json', $sid);
+      if ('POST' === $request->getMethod()) {
+        $this->checkAuthorization();
+        $content = $this->writeFile($path, $request->getContent());
+        return new JsonResponse($content);
+      } else {
+        return new JsonResponse($this->readFile($path), 200, [], true);
+      }
+    } catch (\Throwable $throwable) {
+      return $this->error($throwable);
+    }
+  }
+
+  public function grundeGeojson($sid)
+  {
+    try {
+      $request = $this->requestStack->getCurrentRequest();
+
+      $path = sprintf('%s/grunde.geojson', $sid);
+      if ('POST' === $request->getMethod()) {
+        $this->checkAuthorization();
+        $content = $this->writeFile($path, $request->getContent());
+        return new JsonResponse($content);
+      } else {
+        $contentType = 'application/geo+json';
+        return new JsonResponse($this->readFile($path, $contentType), 200, [
```

Comment:
See above

Reply:
Fixed in https://github.com/grundsalg-aarhus/grundsalg_web/pull/13/commits/58f08684dd856f77e98959d9ab992a570d4d9884.